Add InfluxDB v2 support

raminqaf opened the pull request:
This PR fixes #3670 

Every item on this list will require judgement by the Testcontainers core maintainers. Exceptions will sometimes be possible; items with `should` are more likely to be negotiable than those items with `must`.

#### Default docker image

- [x] Should be a Docker Hub official image, or published by a reputable source (ideally the company or organisation that officially supports the technology)
- [ ] Should have a verifiable open source Dockerfile and a way to view the history of changes
- [x] MUST show general good practices regarding container image tagging - e.g. we do not use `latest` tags, and we do not use tags that may be mutated in the future
- [x] MUST be legal for Testcontainers developers and Testcontainers users to pull and use. Mechanisms exist to allow EULA acceptance to be signalled, but images that can be used without a licence are greatly preferred.

#### Module dependencies

- [x] The module should use as few dependencies as possible, 
- [x] Regarding libraries, either:
    - they should be `compileOnly` if they are likely to already be on the classpath for users' tests (e.g. client libraries or drivers) 
    - they can be `implementation` (and thus transitive dependencies) if they are very unlikely to conflict with users' dependencies.
- [x] If client libraries are used to test or use the module, these MUST be legal for Testcontainers developers and Testcontainers users to download and use.

#### API (e.g. `MyModuleContainer` class)

- [x] Favour doing the right thing, and least surprising thing, by default
- [x] Ensure that method and parameter names are easy to understand. Many users will ignore documentation, so IDE-based substitutes (autocompletion and Javadocs) should be intuitive. 
- [x] The module's public API should only handle standard JDK data types and MUST not expose data types that come from `compileOnly` dependencies. This is to reduce the risk of compatibility problems with future versions of third party libraries.
 
#### Documentation

- [x] Every module MUST have a dedicated documentation page containing:
    - [x] A high level overview
    - [x] A usage example
    - [x] If appropriate, basic API documentation or further usage guidelines
    - [x] Dependency information
    - [x] Acknowledgements, if appropriate
- [ ] Consider that many users will not read the documentation pages - even if the first person to add it to a project does, people reading/updating the code in the future may not. Try and avoid the need for critical knowledge that is only present in documentation.

rnorth commented:
Thanks for the contribution @raminqaf

I think one thing we need to be _really_ careful about is the breaking-change nature of this. Is there no way at all to make the `InfluxDBContainer` be compatible with both v1.x and v2.x of InfluxDB?

We need to avoid a situation where an upgrade of Testcontainers prevents people from being able to run their existing tests that rely on InfluxDB v1.x.

raminqaf commented:
@rnorth unfortunately not. Another way to add support for InfluxDB 2.x and keep the 1.x supported is to add a new and sperated module like `influxDB2`.

rnorth commented:
What if we kept it in the same module, but have a `InfluxDBContainerV2` class? We could also rename the current class to `InfluxDBContainerV1` and have a `@Deprecated class InfluxDBContainer extends InfluxDBContainerV1` to maintain binary compatibility.

raminqaf commented:
@rnorth Sounds good to me! I will add your suggestions and push them to the PR.

raminqaf commented:
@rnorth any updates on this PR?

vcvitaly commented:
I had a quick read and everything is ok at the first glance :eyes:

raminqaf commented:
@vektory79 Thanks! 
Would be could if we could merge this. Since we are using the code in our project. I would like to have it from testcontainers dependency, rather than having it in our project directly :wink: @rnorth @kiview @bsideup

gjemp commented:
hi, 
some updates should be done 
> influxdbV2ClientVersion = '1.14.0'
 there is 2.0.0 out.

the example about v2 is a bit missleading i think cause the default for pure v2 s now token  based . The user and password is only for the v1=>v2 upgraded instances.

raminqaf commented:
> hi,
> some updates should be done
> 
> > influxdbV2ClientVersion = '1.14.0'
> 
> there is 2.0.0 out.
> 
> the example about v2 is a bit missleading i think cause the default for pure v2 s now token based . The user and password is only for the v1=>v2 upgraded instances.

Thanks for the comment and your feedback on the PR. By the time I created this PR the version was `1.14.0` but I will update it to the newest one. The authentication and setup of the InfluxDB could be done either through username and password or through the token. As you can see on the [InfluxDB Docker setup page](https://docs.influxdata.com/influxdb/v2.0/get-started/?t=CLI+Setup#set-up-influxdb-through-the-influx-cli). And according to the [setup flags documentation](https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/setup/#flags) the token is described as: "Token for admin user". Therefore I will keep the authentication through the username and password, which makes it a more realistic scenario for "primary users".

gjemp commented:
> 
> Thanks for the comment and your feedback on the PR. By the time I created this PR the version was `1.14.0` but I will update it to the newest one. The authentication and setup of the InfluxDB could be done either through username and password or through the token. As you can see on the [InfluxDB Docker setup page](https://docs.influxdata.com/influxdb/v2.0/get-started/?t=CLI+Setup#set-up-influxdb-through-the-influx-cli). And according to the [setup flags documentation](https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/setup/#flags) the username and password are still valid arguments to do the authentication. Besides you can use these credentials to log in, during debug mode, to the dashboard and query the data.

Thanks for update. Any idea when we get the review done ? Would really like to have the update  as release.

raminqaf commented:
> 
> > 
> > Thanks for the comment and your feedback on the PR. By the time I created this PR the version was `1.14.0` but I will update it to the newest one. The authentication and setup of the InfluxDB could be done either through username and password or through the token. As you can see on the [InfluxDB Docker setup page](https://docs.influxdata.com/influxdb/v2.0/get-started/?t=CLI+Setup#set-up-influxdb-through-the-influx-cli). And according to the [setup flags documentation](https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/setup/#flags) the username and password are still valid arguments to do the authentication. Besides you can use these credentials to log in, during debug mode, to the dashboard and query the data.
> 
> Thanks for update. Any idea when we get the review done ? Would really like to have the update  as release.

Unfortunately not. In order to merge the PR, one of the three reviewers (@rnorth @bsideup @bsideup) should approved the PR first.

Andy2003 commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,176 @@
+
+    private InfluxDBContainerV2(final DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
```
Running with

```java
InfluxDBContainerV2.createWithSpecificTag(DockerImageName.parse("quay.io/influxdb/influxdb:v2.0.7"));
```

will fail with:

```
aused by: java.lang.IllegalStateException: Failed to verify that image 'quay.io/influxdb/influxdb:v2.0.7' is a compatible substitute for 'quay.io/influxdb/influxdb:v2.0.0'. This generally means that you are trying to use an image that Testcontainers has not been designed to use. If this is deliberate, and if you are confident that the image is compatible, you should declare compatibility in code using the `asCompatibleSubstituteFor` method. For example:
   DockerImageName myImage = DockerImageName.parse("quay.io/influxdb/influxdb:v2.0.7").asCompatibleSubstituteFor("quay.io/influxdb/influxdb:v2.0.0");
and then use `myImage` instead.
	at org.testcontainers.utility.DockerImageName.assertCompatibleWith(DockerImageName.java:248)
	at org.testcontainers.containers.InfluxDBContainerV2.<init>(InfluxDBContainerV2.java:42)
	at org.testcontainers.containers.InfluxDBContainerV2.createWithSpecificTag(InfluxDBContainerV2.java:57)
	at com.kiwigrid.ence.timeseriesservice.config.TestInfluxDbConfig.influxDBContainer(TestInfluxDbConfig.java:21)
```

rnorth replied:
Good spot! `assertCompatibleWith` will look for an exact match on the tag part, if it's set. Usually we should have a `DockerImageName` that has no tag set passed in as a parameter.

If we go with my comment just below about constructors, we'd expect there to be zero occurrences of a tag string in this class.

Andy2003 commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,176 @@
+    /**
+     * @return a url to influxDb
+     */
+    String getUrl() {
```
can you make this public, so we can get the URL?

Andy2003 commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,176 @@
+        this.addEnv("INFLUXDB_ORG", this.organization);
+        this.addEnv("INFLUXDB_RETENTION", String.valueOf(this.retention));
+        this.addEnv("INFLUXDB_RETENTION_UNIT", this.retentionUnit);
+    }
```
Add the ability to set the admin access token as stated here:

https://docs.influxdata.com/influxdb/v2.0/upgrade/v1-to-v2/docker/#influxdb-2x-initialization-credentials

Andy2003 commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,176 @@
+    @Override
+    @SneakyThrows({InterruptedException.class, IOException.class, ExecutionException.class})
+    public void start() {
+        this.withCopyFileToContainer(MountableFile.forClasspathResource(INFLUX_SETUP_SH),
```
is this still required with newer images?

raminqaf replied:
Good point! It is not needed anymore since the InfluxDB image can be configured through the environment variables.
https://hub.docker.com/_/influxdb

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,176 @@
+        this.addExposedPort(INFLUXDB_PORT);
+    }
+
+    public static InfluxDBContainerV2<?> createWithDefaultTag() {
+        return new InfluxDBContainerV2<>(DEFAULT_IMAGE_NAME);
+    }
+
+    public static InfluxDBContainerV2<?> createWithSpecificTag(final DockerImageName imageName) {
+        return new InfluxDBContainerV2<>(imageName);
+    }
```
Please can we remove these? In all Testcontainers container classes, the constructor is our standard way of instantiation, and introducing a new style would be confusing and possibly constrain us when we evolve APIs.

Also we deliberately deprecated the idea of 'default tags' a while back because we found that they would become stale. The preferred style is a public constructor that accepts a `DockerImageName`, which puts the decision about which image to use into the user's hands.

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,176 @@
+        }
+        Startables.deepStart(this.dependencies).get();
+        // trigger LazyDockerClient's resolve so that we fail fast here and not in getDockerImageName()
+        this.dockerClient.authConfig();
```
I don't quite understand why this is here...

raminqaf replied:
I have removed this

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -1,163 +1,10 @@
-        influxDB.setDatabase(database);
-        return influxDB;
+@Deprecated
+public class InfluxDBContainer<SELF extends InfluxDBContainer<SELF>> extends InfluxDBContainerV1<SELF> {
```
It might be helpful to include a Javadoc with a `@deprecated` attribute that instructs people to use either `InfluxDBContainerV1` or `InfluxDBContainerV2` instead.

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -1,163 +1,10 @@
-        return influxDB;
+@Deprecated
+public class InfluxDBContainer<SELF extends InfluxDBContainer<SELF>> extends InfluxDBContainerV1<SELF> {
+    public InfluxDBContainer(final DockerImageName influxdbTestImage) {
```
I think we'd need to copy all the other constructors from `InfluxDBContainerV1` too, for compatibility.

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,164 @@
+/**
+ * See <a href="https://store.docker.com/images/influxdb">https://store.docker.com/images/influxdb</a>
+ */
+class InfluxDBContainerV1<SELF extends InfluxDBContainerV1<SELF>> extends GenericContainer<SELF> {
```
Shouldn't this be public? My thinking is:

* We don't want people using `InfluxDBContainer`, but the reason for that is that the version is ambiguous. It's not because we want to drive people away from using InfluxDB v1 - they might have good reasons for it (like it's part of their production infrastructure that they need to carry on supporting).
* People might want to switch to using `InfluxDBContainerV1` to avoid the deprecation warning, but they need to carry on using InfluxDB v1 for testing.

raminqaf replied:
That is a good point! I will change this to public!

raminqaf commented:
@rnorth and @Andy2003 thanks for the code review! I updated the classes, the README and added your suggestions to the PR.

rnorth commented on docs/modules/databases/influxdb.md:
````diff
@@ -1,8 +1,75 @@
+}
+```
+
+This will create a InfluxDB container with the latest tag (
+available [tags](https://hub.docker.com/_/influxdb?tab=tags&page=1&ordering=last_updated)). The influxDB will be setup
````
Please could we avoid encouraging use of `latest` tags? It results in non-repeatable builds, so is something we prefer to steer people away from.

Also, please could we use `codeinclude` macros for these code snippets, so that they are verified compilable/working?

The [Elasticsearch module docs](/blob/master/docs/modules/elasticsearch.md) are a good example of both of these practices.

rnorth commented on docs/modules/databases/influxdb.md:
````diff
@@ -34,6 +103,3 @@ testImplementation "org.testcontainers:influxdb:{{latest_version}}"
 ```
-
-!!! hint
-    Adding this Testcontainers library JAR will not automatically add a database driver JAR to your project. You should ensure that your project also has a suitable database driver as a dependency.
````
Is this no longer true? I think it is...

rnorth commented on modules/influxdb/build.gradle:
```diff
@@ -1,8 +1,16 @@
 description = "Testcontainers :: InfluxDB"
 
+ext {
+    influxdbV1ClientVersion = '2.21'
+    influxdbV2ClientVersion = '3.1.0'
+}
+
```
Please could we return to the original one-string form for specifying these dependencies? It's the convention we use throughout, and works well with Dependabot.

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,165 @@
+    private String password = "any";
+
+    /**
+     * @deprecated use {@link InfluxDBContainer(DockerImageName)} instead
```
Minor copy/paste issue - I think this should be:
```suggestion
     * @deprecated use {@link InfluxDBContainerV1(DockerImageName)} instead
```

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * @deprecated use {@link InfluxDBContainer(DockerImageName)} instead
```
```suggestion
     * @deprecated use {@link InfluxDBContainerV1(DockerImageName)} instead
```

rnorth commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,165 @@
+        this.addEnv("INFLUXDB_USER_PASSWORD", this.password);
+    }
+
+    @Override
+    public Set<Integer> getLivenessCheckPortNumbers() {
+        return Collections.singleton(this.getMappedPort(INFLUXDB_PORT));
+    }
```
Minor nit, and not at all anything you've done. I think this method is redundant, as the listening port wait strategy.

Let's remove this in a future PR, though!

rnorth commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBV1TestImages.java:
```diff
@@ -2,6 +2,6 @@
 import org.testcontainers.utility.DockerImageName;
 
-public interface InfluxDBTestImages {
+public interface InfluxDBV1TestImages {
```
Small point: could we just have one interface to hold both `DockerImageName` constants?

e.g.
```
DockerImageName INFLUXDB_V1_TEST_IMAGE = DockerImageName.parse("influxdb:1.4.3");
DockerImageName INFLUXDB_V2_TEST_IMAGE = DockerImageName.parse("influxdb:2.0.7");

```

rnorth reviewed (changes requested):
Nice work @raminqaf. I've just made a few more requested changes, mainly related to the docs actually. The actual code looks good to me, mainly!

raminqaf commented:
@rnorth Done! Ready for the merge, hopefully! :smile:

bsideup commented on modules/influxdb/build.gradle:
```diff
@@ -3,6 +3,9 @@ description = "Testcontainers :: InfluxDB"
 
-    compileOnly 'org.influxdb:influxdb-java:2.21'
-    testImplementation 'org.influxdb:influxdb-java:2.21'
+    implementation "org.influxdb:influxdb-java:2.21"
+    implementation "org.influxdb:influxdb-java:2.21"
```
Duplicate:
```suggestion
    implementation "org.influxdb:influxdb-java:2.21"
```

bsideup replied:
Also, previously it was `compileOnly`. @rnorth do we really need to make it a mandatory dependency?

raminqaf replied:
The other one should be `testImplementation`.

rnorth replied:
Actually that's a good point. In the docs we're clear that the user must provide a driver, so `implementation` probably doesn't bring much value.

rnorth commented:
Heads up, running `./gradlew check` locally I'm getting this Javadoc error:
```
/Users/rnorth/github.com/testcontainers/testcontainers-java/modules/influxdb/build/delombok/org/testcontainers/containers/InfluxDBContainerV2.java:60: error: element not closed: a
     * <a href="https://hub.docker.com/_/influxdb">
       ^
1 error
3 warnings
```

I think this will show up in the CI build any moment now...

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,165 @@
+    private String username = "any";
+    private String password = "any";
+
+    /**
+     * @deprecated use {@link InfluxDBContainerV1(DockerImageName)} instead
+     */
+    @Deprecated
+    public InfluxDBContainerV1() {
+        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
+    }
+
```
No need to for it, as this is a new class that wasn't used before
```suggestion
```

rnorth replied:
It's called by `InfluxDBContainer()`

raminqaf replied:
Agree with @rnorth

bsideup replied:
@rnorth `InfluxDBContainer()` can call `InfluxDBContainerV1(DockerImageName)`, as constructors are not inherited

raminqaf replied:
@bsideup can you please help on this one? I don't really understand. Should I explicitly create a new Object in the `InfluxDBContainer()` class?

bsideup replied:
@raminqaf you can do it like this:
```java
    /**
     * @deprecated use {@link InfluxDBContainer(DockerImageName)} instead
     */
    @Deprecated
    public InfluxDBContainer() {
         this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
     }

     /**
      * @deprecated use {@link InfluxDBContainer(DockerImageName)} instead
      */
     @Deprecated
     public InfluxDBContainer(final String version) {
         this(DEFAULT_IMAGE_NAME.withTag(version));
     }
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,165 @@
+        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
+    }
+
+    /**
+     * @deprecated use {@link InfluxDBContainerV1(DockerImageName)} instead
+     */
+    @Deprecated
+    public InfluxDBContainerV1(final String version) {
+        this(DEFAULT_IMAGE_NAME.withTag(version));
+    }
```
Same here - consider making it a non-deprecated `String imageName` constructor

raminqaf replied:
So should I replace this constructor with `public InfluxDBContainerV1(final String imageName)`? This would cause the `InfluxDBContainer()` to fail. Same as the previous constructor you suggested to delete

raminqaf replied:
The reason we kept these constructors is to maintain binary compatibility.

bsideup replied:
constructors are not inherited in Java - you need to explicitly define them again in `InfluxDBContainer`

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,165 @@
+/**
+ * See <a href="https://store.docker.com/images/influxdb">https://store.docker.com/images/influxdb</a>
+ */
+public class InfluxDBContainerV1<SELF extends InfluxDBContainerV1<SELF>> extends GenericContainer<SELF> {
```
Self typing is not needed here - we should take this as an opportunity to change it

```suggestion
public class InfluxDBContainerV1 extends GenericContainer<InfluxDBContainerV1> {
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,166 @@
+ * <a href="https://hub.docker.com/_/influxdb"> the official InfluxDB 2.x container repository</a>
+ * on docker hub for detailed documentation and newest tags.
+ */
+public class InfluxDBContainerV2<SELF extends InfluxDBContainerV2<SELF>> extends GenericContainer<SELF> {
```
Not needed
```suggestion
public class InfluxDBContainerV2 extends GenericContainer<InfluxDBContainerV2> {
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,166 @@
+    private Optional<String> adminToken = Optional.empty();
+
+
+    public InfluxDBContainerV2(final DockerImageName imageName) {
```
Consider adding `InfluxDBContainerV2(String imageName)` for convenience

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,166 @@
+    public InfluxDBContainerV2(final DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        this.waitStrategy = (new WaitAllStrategy())
+            .withStrategy(Wait
+                .forHttp("/ping")
+                .withBasicCredentials(this.username, this.password)
+                .forStatusCode(NO_CONTENT_STATUS_CODE))
+            .withStrategy(Wait.forListeningPort());
```
```suggestion
        this.waitStrategy = new WaitAllStrategy()
            .withStrategy(
                Wait
                    .forHttp("/ping")
                    .withBasicCredentials(this.username, this.password)
                    .forStatusCode(NO_CONTENT_STATUS_CODE)
            )
            .withStrategy(Wait.forListeningPort());
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,166 @@
+     * @see InfluxDBClientOptions
+     * @see InfluxDBClient
+     */
+    public InfluxDBClient getInfluxDBClient() {
```
@rnorth we really shouldn't be adding 3rd party types to our APIs anymore 😅

rnorth replied:
It's a pain, but it was already there in the original `InfluxDBContainer` implementation. This is why I think it's too late to change.

bsideup replied:
Well, it is never too late to change in a newly added code :D
In fact, even V1 IMO should not have it, only now-deprecated `InfluxDBContainer`.

raminqaf replied:
@bsideup so you suggest removing this method and let the user build its own client each time?

bsideup replied:
Yes. See https://www.testcontainers.org/contributing/#module-dependencies

raminqaf replied:
@bsideup and @rnorth I removed the `getInflucDBClient()` function and there is no InfluxDB client dependency for the `InfluxDBV2` anymore.

bsideup commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1Test.java:
```diff
@@ -1,43 +1,44 @@
 package org.testcontainers.containers;
 
-import org.influxdb.InfluxDB;
```
imports are wrong here. Static imports should go after non-static ones.

bsideup commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1WithUserTest.java:
```diff
@@ -1,44 +1,43 @@
 package org.testcontainers.containers;
 
+import static org.hamcrest.CoreMatchers.hasItem;
```
imports are wrong here. Static imports should go after non-static ones.

bsideup commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV2Test.java:
```diff
@@ -0,0 +1,66 @@
+package org.testcontainers.containers;
+
+import static org.hamcrest.CoreMatchers.is;
```
imports are wrong here. Static imports should go after non-static ones.

bsideup commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV2WithUserTest.java:
```diff
@@ -0,0 +1,93 @@
+package org.testcontainers.containers;
+
+import static org.hamcrest.CoreMatchers.is;
```
imports are wrong here. Static imports should go after non-static ones.

bsideup commented on modules/influxdb/build.gradle:
```diff
@@ -3,6 +3,8 @@ description = "Testcontainers :: InfluxDB"
 
-    compileOnly 'org.influxdb:influxdb-java:2.21'
-    testImplementation 'org.influxdb:influxdb-java:2.21'
+    implementation "org.influxdb:influxdb-java:2.21"
+    testImplementation "org.influxdb:influxdb-java:2.21"
```
Let's revert this change completely
```suggestion
    compileOnly 'org.influxdb:influxdb-java:2.21'
    testImplementation 'org.influxdb:influxdb-java:2.21'
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -1,163 +1,22 @@
-
 /**
- * See <a href="https://store.docker.com/images/influxdb">https://store.docker.com/images/influxdb</a>
+ * @deprecated instead use {@link InfluxDBContainerV1#InfluxDBContainerV1(DockerImageName) } for InfluxDB 1.x or {@link
+ * InfluxDBContainerV2#InfluxDBContainerV2(DockerImageName)} for InfluxDB 2.x instead
```
```suggestion
 * @deprecated instead use {@link InfluxDBContainerV1} for InfluxDB 1.x or {@link InfluxDBContainerV2} for InfluxDB 2.x instead
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,165 @@
+
+
+    public InfluxDBContainerV2(final String imageName) {
+        super(DockerImageName.parse(imageName));
+        final DockerImageName dockerImageName = DockerImageName.parse(imageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        this.setWaitStrategy();
```
```suggestion
        this(DockerImageName.parse(imageName));
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,165 @@
+    public InfluxDBContainerV2(final DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        this.setWaitStrategy();
```
with the suggestion above, you can inline `setWaitStrategy` here

bsideup commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1WithUserTest.java:
```diff
@@ -1,44 +1,43 @@
 import org.junit.Rule;
 import org.junit.Test;
 
-import java.util.concurrent.TimeUnit;
```
Could you please revert this change? `java.*` imports should go after non-JDK ones, not before

raminqaf commented:
@bsideup and @rnorth piiiiiuuuhhhhh :sweat_smile: all done!

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -1,35 +1,15 @@
-    public static final Integer INFLUXDB_PORT = 8086;
-
+@Deprecated
+public class InfluxDBContainer extends InfluxDBContainerV1 {
```
To make it binary compatible, we need to keep the self typing on this class:
```suggestion
public class InfluxDBContainer<SELF extends InfluxDBContainer<SELF>> extends InfluxDBContainerV1 {
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java:
```diff
@@ -0,0 +1,159 @@
+ */
+public class InfluxDBContainerV2 extends GenericContainer<InfluxDBContainerV2> {
+
+    public static final Integer INFLUXDB_PORT = 8086;
```
```suggestion
    private static final Integer INFLUXDB_PORT = 8086;
```

bsideup commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV1.java:
```diff
@@ -0,0 +1,150 @@
+
+    private static final int NO_CONTENT_STATUS_CODE = 204;
+
+    @Deprecated
+    public static final String VERSION = DEFAULT_TAG;
```
```suggestion
    static final String VERSION = DEFAULT_TAG;
```

raminqaf commented:
@bsideup I tried to understand your suggestions... Please check if it is fine :smile:

bsideup commented:
@raminqaf we shouldn't be changing the `V1` class, but instead override its methods and return `SELF` from them (with something like `return (SELF) this;`)

raminqaf commented:
@rnorth and @bsideup Is the PR ready for the merge?

eddumelendez commented:
@raminqaf thank for your patience on this one! we are still interested on getting this merge. However, a few things have changed since last time and we believe nowadays we can have one class container instead of two. Looks like the the only the difference are in the env variables. We can do now check which version image is used and set those variables according to the version. Check [elasticsearch](https://github.com/testcontainers/testcontainers-java/blob/master/modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java#L102-L103) for inspiration. Please, let us know if you are still interested on work on this one and if you prefer to update the PR or I think would be much easier to create a new PR.

raminqaf commented:
@eddumelendez Hello! I am still interested in this PR and would love to work on it again! It's been a while since I pushed on this branch! 😅 I need to get into it and will look at the Elasticsearch code for that! Thanks!

raminqaf commented:
@eddumelendez I have added the changes you requested. Unfortunately, I cannot build the project locally on my laptop (Apple m1 Pro). I am getting this error every time I want to build the project:
```
Build file 'testcontainers/build.gradle' line: 65

Could not compile build file 'testcontainers/build.gradle'.
> startup failed:
  build file 'testcontainers/build.gradle': 65: unable to resolve class org.testcontainers.build.DelombokArgumentProvider 
   @ line 65, column 13.
                 new org.testcontainers.build.DelombokArgumentProvider(srcDirs: project.sourceSets.main.java.srcDirs, outputDir: file("$buildDir/delombok"))
```
I still like to update the InfluxDB Java client to the newest version for the tests. But first, I need to solve this build problem.

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -1,83 +1,163 @@
     protected void configure() {
-        addEnv("INFLUXDB_ADMIN_USER", admin);
-        addEnv("INFLUXDB_ADMIN_PASSWORD", adminPassword);
+        if (isAtLeastMajorVersion2) {
```
latest should be included as a version 2?

raminqaf replied:
According to the [Dockerhub description](https://hub.docker.com/_/influxdb?tab=description), yes:
```
The latest tag for this image now points to the latest released implementation of InfluxDB 2.x. If you are using the latest tag and would like to stay on the InfluxDB 1.x line, please update your environment to reference the 1.8 tag.
```

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -120,48 +200,68 @@ public SELF withAdminPassword(final String adminPassword) {
-     * @return a influxDb client
+     * @return a InfluxDB client for InfluxDB &lt;= 2.
      */
     public InfluxDB getNewInfluxDB() {
```
deprecate this method please

eddumelendez commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1WithUserTest.java:
```diff
@@ -0,0 +1,75 @@
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+public class InfluxDBContainerV1WithUserTest {
```
let's move those tests to `InfluxDBContainerV1Test` please

eddumelendez commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1Test.java:
```diff
@@ -0,0 +1,46 @@
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class InfluxDBContainerV1Test {
```
instead of using `@ClassRule`. Let's use `try-with-resources` in each test method and create the container in there.

eddumelendez commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerWithUserTest.java:
```diff
@@ -1,75 +1,94 @@
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.junit.Assert.assertThat;
 
 public class InfluxDBContainerWithUserTest {
```
those tests can be moved to `InfluxDBContainerTest`

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,22 +1,53 @@
+<!--/codeinclude-->
+
+
+The influxDB will be setup with the following data:<br/>
+
+| Property      | Default Value | 
+| ------------- |:-------------:|
+| username      | test-user     | 
+| password      | test-password | 
+| organization  | test-org      |
+| bucket        | test-bucket   |  
+| retention     | 0 (infinite)  |
+| adminToken    |       -       |
```
This only applies to v2, right? we should consider to give visibility about what is set up for each version

eddumelendez commented on docs/modules/databases/influxdb.md:
````diff
@@ -1,22 +1,53 @@
-         InfluxDB influxDB = influxDbContainer.getNewInfluxDB();
-         ...
-```
+<!--codeinclude-->
+[InfluxDBContainerTest](../../../modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java)
+<!--/codeinclude-->
````
we should consider adding a snippet on how to build a client using v1 and v2

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,22 +1,53 @@
-## Usage example
+## Important Note
+
+They are breaking changes in InfluxDB 2.x. For more information refer to the
```
```suggestion
There are breaking changes in InfluxDB 2.x. For more information refer to the
```

eddumelendez reviewed (changes requested):
thanks for updating the PR @raminqaf ! I have added more comments but very glad to see is getting shape :)

raminqaf commented:
@eddumelendez I added all the reviews you asked for! I hope the PR is ready!

raminqaf commented:
> @eddumelendez I have added the changes you requested. Unfortunately, I cannot build the project locally on my laptop (Apple m1 Pro). I am getting this error every time I want to build the project:
> 
> ```
> Build file 'testcontainers/build.gradle' line: 65
> 
> Could not compile build file 'testcontainers/build.gradle'.
> > startup failed:
>   build file 'testcontainers/build.gradle': 65: unable to resolve class org.testcontainers.build.DelombokArgumentProvider 
>    @ line 65, column 13.
>                  new org.testcontainers.build.DelombokArgumentProvider(srcDirs: project.sourceSets.main.java.srcDirs, outputDir: file("$buildDir/delombok"))
> ```
> 
> I still like to update the InfluxDB Java client to the newest version for the tests. But first, I need to solve this build problem.

@eddumelendez Maybe this is interesting (and important) for you: I couldn't solve this problem on my MacBook! I switched to my old Linux laptop to be able to build and test the project!

eddumelendez commented:
First of all, thank you for taking care of the comments! intel chip or m1? I currently use m1 and everything is fine. Used to use intel chip months ago and it worked fine too, as far as I remember but would give it a try again and see what's going on. Thanks for let us know

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -1,39 +1,56 @@
 
 /**
- * See <a href="https://store.docker.com/images/influxdb">https://store.docker.com/images/influxdb</a>
+ * Represents an InfluxDB Docker instance.
+ * Refer to
+ * <a href="https://hub.docker.com/_/influxdb"> the official InfluxDB 1.x and 2.x container repository</a>
+ * on docker hub for detailed documentation and newest tags.
  */
```
```suggestion
 * Testcontainers implementation for InfluxDB.
 */
```

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,40 +82,122 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
     }
 
+    /**
+     * Set env variable `INFLUXDB_USER` for InfluxDB &lt;= 2.
+     * <p>
+     * Set env variable `DOCKER_INFLUXDB_INIT_USERNAME` for InfluxDB &gt;= 2.
```
I think it should be INFLUXDB_USER for < 1 and DOCKER_INFLUXDB_INIT_USERNAME for >= 2, right? same with `withPassword`.

eddumelendez replied:
@raminqaf 👀

raminqaf replied:
@eddumelendez, for me it would make sense to make it like this:
```
I think it should be INFLUXDB_USER for < 2 and DOCKER_INFLUXDB_INIT_USERNAME for >= 2, right? same with withPassword.
```
because the version less than 1 does not sound correct to me.
Or the other variation would be like this:
```
I think it should be INFLUXDB_USER for 1.x and DOCKER_INFLUXDB_INIT_USERNAME for 2.x, right? same with withPassword.
```
I like the second one more. Because we have the same versioning model in the readme as well.

raminqaf commented:
> First of all, thank you for taking care of the comments! intel chip or m1? I currently use m1 and everything is fine. Used to use intel chip months ago and it worked fine too, as far as I remember but would give it a try again and see what's going on. Thanks for let us know

I have a MacBook with the M1 chip.

raminqaf commented:
@eddumelendez Is it ready? 🚀

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,40 +82,122 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
     }
 
+    /**
+     * Set env variable `INFLUXDB_USER` for InfluxDB 1.x.
+     * <p>
+     * Set env variable `DOCKER_INFLUXDB_INIT_USERNAME` for InfluxDB 2.x.
```
Let's not provide much detail about implementation detail in javadocs

```suggestion
     * Set user for InfluxDB.
```

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,40 +82,122 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
+    }
+
+    /**
+     * Set env variable `INFLUXDB_USER_PASSWORD` for InfluxDB 1.x.
+     * <p>
+     * Set env variable `DOCKER_INFLUXDB_INIT_PASSWORD` for InfluxDB 2.x.
```
```suggestion
     * Set password for InfluxDB.
```

eddumelendez commented on docs/modules/databases/influxdb.md:
````diff
@@ -1,40 +1,111 @@
+In the following you will find a snippet to create a InfluxDB client using the java client:
+
+```java
+class InfluxDBClient {
+    public static InfluxDBClient getInfluxDBClient(final InfluxDBContainer influxDBContainer) {
+        final InfluxDB influxDB = InfluxDBFactory.connect(influxDBContainer.getUrl(), influxDBContainer.getUsername(), influxDBContainer.getPassword());
+        influxDB.setDatabase(influxDBContainer.getDatabase());
+        return influxDB;
+    }
+}
````
use code snippets from tests

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,40 +1,111 @@
-    public void someTestMethod() {
-         InfluxDB influxDB = influxDbContainer.getNewInfluxDB();
-         ...
+class InfluxDBClient {
+    public static InfluxDBClient getInfluxDBClient(final InfluxDBContainer influxDBContainer) {
+        final InfluxDBClientOptions influxDBClientOptions = InfluxDBClientOptions.builder()
+            .url(influxDBContainer.getUrl())
+            .authenticate(influxDBContainer.getUsername(), influxDBContainer.getPassword().toCharArray())
+            .bucket(influxDBContainer.getBucket())
+            .org(influxDBContainer.getOrganization())
+            .build();
+        return InfluxDBClientFactory.create(influxDBClientOptions);
+    }
+}
```
use code snippets from tests

eddumelendez reviewed (approved):
Thank you so much @raminqaf !

kiview commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -97,21 +202,21 @@ public SELF withAuthEnabled(final boolean authEnabled) {
     }
 
     /**
      * Set env variable `INFLUXDB_ADMIN_PASSWORD`.
```
```suggestion
     * Set admin password
```

kiview commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -120,48 +225,70 @@ public SELF withAdminPassword(final String adminPassword) {
 
     /**
-     * Set env variable `INFLUXDB_USER`.
+     * Set env variable `DOCKER_INFLUXDB_INIT_ORG`.
```
```suggestion
     * Set organization name
```

kiview commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -120,48 +225,70 @@ public SELF withAdminPassword(final String adminPassword) {
 
     /**
-     * Set env variable `INFLUXDB_USER_PASSWORD`.
+     * Set env variable `DOCKER_INFLUXDB_INIT_BUCKET`.
```
```suggestion
     * Initialize bucket with given name
```

kiview commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -120,48 +225,70 @@ public SELF withAdminPassword(final String adminPassword) {
 
     /**
-     * @return a url to influxDb
+     * Set env variable `DOCKER_INFLUXDB_INIT_RETENTION`.
```
```suggestion
     * Set the retention in days
```
Is this retention in days?

kiview commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -120,48 +225,70 @@ public SELF withAdminPassword(final String adminPassword) {
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_ADMIN_TOKEN`.
```
```suggestion
     * Set admin token
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+    @Nullable
+    private InfluxDBClient influxDBClient = null;
+
+    @After
+    public void stopInfluxDBClient() {
+        if (this.influxDBClient != null) {
+            this.influxDBClient.close();
+            this.influxDBClient = null;
+        }
+    }
```
Couldn't we add `influxDBClient` to the try-with-resources blocks instead?

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
+        ) {
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
```
The `isRunning()` asserts should be superfluous, the container should always be running after successfully proceeding after a `start()` call.

kiview replied:
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
 
-        assertThat(actual).isNotNull();
+            final String actual = influxDBContainer.getUrl();
+            assertThat(actual).isNotNull();
```
Let's try to have a higher level of abstraction in the tests. E.g. in this case, do something similar to the `ping()` call in the other test, but construct the client using the returned URL.

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
+        ) {
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
```
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
 
-        assertThat(actual).isNotNull();
-        assertThat(actual.ping()).isNotNull();
+            assertThat(this.influxDBClient).isNotNull();
```
```suggestion
```

I don't think we need these redundant assertions, makes the test IMO harder to read.

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+            assertThat(this.influxDBClient).isNotNull();
+
+            final Bucket bucket = this.influxDBClient.getBucketsApi().findBucketByName(BUCKET);
+            assertThat(bucket).isNotNull();
```
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+
+            this.influxDBClient = InfluxDBTestUtils.getInfluxDBClient(influxDBContainer);
+
+            assertThat(this.influxDBClient).isNotNull();
```
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
 
-        assertThat(actual).isNotNull();
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
```
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+                .withAdminToken(ADMIN_TOKEN)
+        ) {
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
```
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
+
+            this.influxDBClient = InfluxDBTestUtils.getInfluxDBClient(influxDBContainer);
+
+            assertThat(this.influxDBClient).isNotNull();
```
```suggestion
```

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,137 @@
     @Test
-    public void isRunning() {
-        boolean actual = influxDBContainer.isRunning();
+    public void queryForWriteAndRead() {
```
I don't understand the idea of this test. Is there anything in the implementation code that would make this test fail, but would still keep all other tests green?

raminqaf replied:
IMO this test ensures that the written values are read correctly and all the setup is done right.

raminqaf replied:
I would keep it

kiview commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1Test.java:
```diff
@@ -0,0 +1,151 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class InfluxDBContainerV1Test {
```
Same comments from V2 class apply here.

kiview reviewed (changes requested):
Thanks for the module and the great collaboration @raminqaf.  I added a final review from my side, regarding the structure of some of the tests and specifically about not leaking implementation details into the documentation.

raminqaf commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,40 +82,118 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
 
-        waitStrategy =
+        this.logger().info("Starting an InfluxDB container using [{}]", dockerImageName);
+        this.waitStrategy =
```
Good Point! I didn't know that `HttpWaitStrategy` existed!

kiview replied:
By using `Wait.forHttp()`, you were already using a convenience builder for constructing the `HttpWaitStrategy` here 🙂

raminqaf commented:
@kiview Is this good to merge?

eddumelendez commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,124 @@
-    public void getLivenessCheckPort() {
-        Integer actual = influxDBContainer.getLivenessCheckPort();
+    public void getBucket() {
+        try (
+            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
+        ) {
+            influxDBContainer
+                .withUsername(USERNAME)
+                .withPassword(PASSWORD)
+                .withOrganization(ORG)
+                .withBucket(BUCKET)
+                .withRetention(RETENTION);
```
```suggestion
        try (
            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
                .withUsername(USERNAME)
                .withPassword(PASSWORD)
                .withOrganization(ORG)
                .withBucket(BUCKET)
                .withRetention(RETENTION);
        ) {
                
```

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,40 +1,93 @@
+In the following example you will find a snippet to create an InfluxDB client using the official Java client:
+
+<!--codeinclude-->
+[InfluxDBTestUtils.getInfluxDBClient()](../../../modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBTestUtils.java)
```
same here about use the code that is required

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,40 +1,93 @@
+Running a `InfluxDBContainer` as a stand-in for InfluxDB in a test:
+
+<!--codeinclude-->
+[InfluxDBContainerV1Test](../../../modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1Test.java)
```
same here about use the code that is required

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,113 +80,195 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
      */
     public String getUrl() {
-        return "http://" + getHost() + ":" + getLivenessCheckPort();
+        return "http://" + this.getHost() + ":" + this.getMappedPort(INFLUXDB_PORT);
```
```suggestion
        return "http://" + getHost() + ":" + getMappedPort(INFLUXDB_PORT);
```

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,113 +80,195 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
     public InfluxDB getNewInfluxDB() {
-        InfluxDB influxDB = InfluxDBFactory.connect(getUrl(), username, password);
-        influxDB.setDatabase(database);
+        final InfluxDB influxDB = InfluxDBFactory.connect(this.getUrl(), this.username, this.password);
```
```suggestion
        final InfluxDB influxDB = InfluxDBFactory.connect(getUrl(), this.username, this.password);
```

eddumelendez commented on modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java:
```diff
@@ -55,113 +80,195 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
+            .withBasicCredentials(this.username, this.password)
+            .forStatusCode(NO_CONTENT_STATUS_CODE);
+
+        this.isAtLeastMajorVersion2 =
+            new ComparableVersion(dockerImageName.getVersionPart()).isGreaterThanOrEqualTo("2.0.0");
```
@kiview I wonder if we can remove this check and use the version provided via ENV [here](https://github.com/influxdata/influxdata-docker/blob/8a3a2ae5bdae946cac8371a69cad45600830224f/influxdb/1.8/Dockerfile#L12). In order to do this we need to make a `dockerClient.inspectImageCmd(image)` at the configure method. WDYT?

eddumelendez replied:
I will leave it as is for now. I think we need to think about how to address them for custom images.

eddumelendez reviewed (changes requested):
@raminqaf thanks once again! I just noticed some important changes that we should address before merge.

raminqaf commented:
@eddumelendez Thanks for the review! I added all of them!

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,40 +1,93 @@
+Running a `InfluxDBContainer` as a stand-in for InfluxDB in a test:
+
+<!--codeinclude-->
+[InfluxDBContainerTest](../../../modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java)
```
this one is missing

raminqaf replied:
@eddumelendez Done!

eddumelendez commented on docs/modules/databases/influxdb.md:
```diff
@@ -1,40 +1,93 @@
+In the following example you will find a snippet to create an InfluxDB client using the official Java client:
+
+<!--codeinclude-->
+[InfluxDBContainer.getNewInfluxDB()](../../../modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java)
```
missing the block here too

raminqaf replied:
Done ✅

eddumelendez commented:
It's failing because of

```
Error: eckstyle] [ERROR] /home/runner/work/testcontainers-java/testcontainers-java/modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:32:5: 'VARIABLE_DEF' should be separated from previous line. [EmptyLineSeparator]
Error: eckstyle] [ERROR] /home/runner/work/testcontainers-java/testcontainers-java/modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerV1Test.java:4:62: Using a static member import should be avoided - org.testcontainers.containers.InfluxDBTestUtils.createInfluxDBWithUrl. [AvoidStaticImport]
```

You can reproduce it locally by running `./gradlew :influxdb:checkstyleMain :influxdb:checkstyleTest`

eddumelendez commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,135 @@
+        try (
+            // constructorWithDefaultVariables {
+            final InfluxDBContainer<?> influxDBContainer = new InfluxDBContainer<>(
+                InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
```
code snippets are using `InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE` which is not useful for docs. Just in those cases can we use the string image name, please? Same for `InfluxDBTestUtils.INFLUXDB_V1_TEST_IMAGE`

eddumelendez commented on modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java:
```diff
@@ -1,44 +1,135 @@
+
+    private static final String RETENTION = "1w";
+
+    private static final String ADMIN_TOKEN = "super-secret-token";
+    private static final int SECONDS_IN_WEEK = 604800;
```
```suggestion
    private static final String ADMIN_TOKEN = "super-secret-token";

    private static final int SECONDS_IN_WEEK = 604800;
```

eddumelendez reviewed (changes requested):
thanks a lot for your patience @raminqaf ! I think we are just polishing the docs now and from my side LGTM.

raminqaf commented:
@eddumelendez I added all the reviews and the checkstyle checks!

eddumelendez commented:
thank you so much for your contribution and all work you've done, @raminqaf ! This is now in `main` branch and will be available in next version.
